src/web/views/Content/Select/index.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+import { Modal, Button, Radio, OverlayTrigger, Tooltip, Panel, Well } from 'react-bootstrap'
+import Select from 'react-select'
+import axios from 'axios'
+import classnames from 'classnames'
+
+import CreateModal from '../modal'
+import {
+  fetchContentMessagesRecent,
+  fetchContentMessagesCount,
+  fetchContentCategories,
+  upsertContentMessages
+} from '~/actions'
+
+const style = require('./style.scss')
+
+class SelectContent extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = { show: false, category: null, searchTerm: '', contentToEdit: null }
+
+    window.botpress = window.botpress || {}
+    window.botpress.pickContent = (options = {}, callback) => {
+      this.props.fetchContentMessagesRecent({})
+      this.props.fetchContentMessagesCount()
+      this.props.fetchContentCategories()
+      this.setState({ show: true, callback })
```

Comment:
Only store in the state what is required to visually render the component (in this case `callback` isn't needed)

Reply:
Right, callback can go on `this`
But in general why do we need to expose it on global instance? This kinda counter-Reactive?

Reply:
That's coming from my spec, I wanted a way to make it globally available to all modules, though we could have passed them as a prop for sure

Reply:
So here's my suggestion, actually multiple
Common part: make this a pure component, it should be controlled, `show` coming from `props`, `onClose` and `onSelected` being porops, too. There's a single instance of Selector, just as now, injected on the global level. Not sure why it leaves in the Sidebar, probably better to keep it on the top level, in App or Layout?

Then there are variations:
* option 1: we still make the global method, but it's created outside of this component, somewhere in index.jsx or lite.jsx? It will save the callback and `showContentSelect` in the high level comp's state, and pass them to the selector as props
* option 2: we will not expose this method globally and pass to any 3rd party module as a props
* option 3, the most flexible: we will pas the component class itself as the prop instead, and may eventually pass more useful classes that way

@slvnperron what do you think as the author of the original spec?

Reply:
Thanks Eugene. Agreed it should sit at the App/Layout level instead of the Sidebar (I didn't catch that). Did you mean it should be a smart component (i.e. connected in redux's terms)?

I think option 3 is the best, we just need to figure out if we'll need React's upcoming Portals as 3rd party modules are not mounted on the same DOM node. Might be a bit complicated, in which case we can go option 2.

Reply:
No I didn't mean for it to be smart, instead I suggest for it to be completely controlled (show is passed as props, onClosed is from props, etc)
For Portals you only need an element so you can also inject the `renderRoot` as a props to modules and let them use it for their portals if needed (but actually if the component _itself_ renders to the portal it shouldn't matter where the caller is mounted, or am I wrong?)

Reply:
Oh, it can keep fetching them as now, I mean no changes to that.
If you want this `pickContent` convenience method I'm also fine with that, but the component itself should not inject anything to the outer world, the modules injection can own the instance of this component and provide this prop to modules

Reply:
Oh, got you. Yes, agreed.